UMA-Bots/Large-Relayer-Token-Balance-Change/src/agent.spec.ts

```diff
@@ -0,0 +1,138 @@
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a transfer is detected from a monitored address for a non-monitored token", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[1])
+      .addEventLog(TRANSFER_EVENT, RANDOM_ADDRESSES[0], [
+        TEST_MONITORED_ADDRESS,
+        RANDOM_ADDRESSES[1],
+        RANDOM_INTEGERS[0],
+      ]);
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a transfer is detected from a non-monitored address for a monitored token", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[1])
+      .addEventLog(TRANSFER_EVENT, MONITORED_ERC20_ADDR, [
+        RANDOM_ADDRESSES[0],
+        RANDOM_ADDRESSES[1],
+        RANDOM_INTEGERS[0],
+      ]);
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns a finding if a monitored address acquires monitored tokens with initial zero balance", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(TRANSFER_EVENT, MONITORED_ERC20_ADDR, [RANDOM_ADDRESSES[0], TEST_MONITORED_ADDRESS, "1000000"]);
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([testGetFindingInstance("1000000", TEST_MONITORED_ADDRESS, "true")]);
+  });
+
+  it("returns a finding if a monitored address receives a large (more than 50% change) amount of tokens", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(TRANSFER_EVENT, MONITORED_ERC20_ADDR, [RANDOM_ADDRESSES[0], TEST_MONITORED_ADDRESS, "500000"]);
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([testGetFindingInstance("500000", TEST_MONITORED_ADDRESS, "true")]);
+  });
+
+  it("returns a finding if a monitored address sends a large (more than 50% change) amount of tokens", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(TRANSFER_EVENT, MONITORED_ERC20_ADDR, [TEST_MONITORED_ADDRESS, RANDOM_ADDRESSES[0], "1000000"]);
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([testGetFindingInstance("1000000", TEST_MONITORED_ADDRESS, "false")]);
+  });
+
+  it("returns N finding only for those transactions that cross the percentage threshold (for monitored address and monitored token) when N > 1", async () => {
+    // current balance in testLru for MONITORED_ERC20_ADDR TEST_MONITORED_ADDRESS is 1500000
+    // current balance in testLru for MONITORED_ERC20_ADDR TEST_MONITORED_ADDRESS is 0
```

Comment:
```suggestion
```

The tests should be independent of each other, so I think you need to add a `testLru.clear()` in a `beforeEach` block. You could add a 2nd tx event where it is needed to add the balance in a previous block

Reply:
@Vxatz, I am pushing a commit that solves the issue you mentioned, but it is in a bit different manner. I am simply initializing the lru cache each time in the `beforeEach` block. For the last test case (where adding some imaginary context is needed), I am setting the values in testLru to the imaginary context. Please give it a look and see if I should change it nevertheless.